src/shapes/SurfacePolygon.js

```diff
@@ -116,5 +116,39 @@ define([
         SurfacePolygon.prototype.computeBoundaries = function(dc) {
         };
 
+        // Internal use only. Intentionally not documented.
+        SurfacePolygon.prototype.getReferencePosition = function () {
+            // Assign the first position as the reference position.
+            if(this.boundaries.length > 0 && this.boundaries[0].length > 2){
+                return this.boundaries[0][0];
+            }
+            else if (this.boundaries.length > 2){
+                return this.boundaries[0];
+            }
+            else {
+                return null;
+            }
+        };
+
+        // Internal use only. Intentionally not documented.
+        SurfacePolygon.prototype.moveTo = function (globe, position) {
+            if(this.boundaries.length > 0 && this.boundaries[0].length > 2){
+                var boundaries = [];
+                for (var i = 0; i < this._boundaries.length; i++){
```

Comment:
Here is the same issue in the for loop as mentioned for `computeShiftedLocations`. Please improve it as well with `for(var i = 0, len = this._boundaries.length; i < len; i++)`.